### ISSUE TITLE
Not all parameters exposed in McEliece?

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [PQClean/PQClean#518]
The McEliece cryptosystem fixes _n_, _k_, _t_, _f_, and _F_ (originally "_g_") as the parameters 2 parties have to agree on to inter-operate.

For example, the 6960119 parameter set [fixes](https://classic.mceliece.org/nist/mceliece-20201010.pdf#subsection.3.7) _n_ = 6960, _k_ = 5413 (from _k_ = _n_ − _td_, _d_ = 13); _t_ = 119, _f_ = `{ 13 , 4 , 3 , 1 , 0 }`  (8219 dec), and _F_ = `{ 119 , 8 , 0 }` (664613997892457936451903530140172545 dec).

However, of these 5 parameters, I can only seem to programmatically extract 3:

* _n_ can be found at [`params.h@SYS_N`](https://github.com/PQClean/PQClean/blob/da93bbc4c4e4356bee6ace5cac32ac67b1f6f179/crypto_kem/mceliece6960119/clean/params.h#L7)
* _k_ can be derived as _k_ = _n_ − _td_
  * _d_ can be found at [`params.h@GFBITS`](https://github.com/PQClean/PQClean/blob/da93bbc4c4e4356bee6ace5cac32ac67b1f6f179/crypto_kem/mceliece6960119/clean/params.h#L6)
* _t_ can be found at [`params.h@SYS_T`](https://github.com/PQClean/PQClean/blob/da93bbc4c4e4356bee6ace5cac32ac67b1f6f179/crypto_kem/mceliece6960119/clean/params.h#L8)
* _f_ seemingly can't be found; `grep -E '(13.*4.*3.*1|8219)' -r crypto_kem/mceliece6960119` returned no relevant results.
* _F_ seemingly can't be found; `grep -E '(119.*8.*0|664613997892457936451903530140172545)' -r crypto_kem/mceliece6960119` returned no results.

How can these parameters be programmatically extracted?

**Comment by thomwiggers** [comment]
As we have only adapted the upstream source code, this is a question for the Classic McEliece submission team.

**Comment by tniessen** [comment]
I have implemented McEliece bindings based on the reference implementation (https://github.com/tniessen/node-mceliece-nist) and based on this repository (https://github.com/tniessen/node-pqclean), and I don't think either exposed anything besides _k_, _m_, _n_, and _t_.

I believe most modern standards prefer named, standardized parameter sets over explicitly encoded parameters (e.g., MODP groups or elliptic curves). In this case, you might not need to explicitly refer to _f_ or _F_ if the protocol agrees on the name of the parameter set that fixes _f_ and _F_.

**Comment by thomwiggers** [comment]
Agreed, these values should be mostly viewed as implementation details and are not up for negotiation.

**Comment by JamesTheAwesomeDude** [comment]
> In this case, you might not need to explicitly refer to f or F if the protocol agrees on the **name** of the parameter set...

Yeah, named parameters are another option, since they are only going to be of a set of hard-coded constants in general use...

**Comment by JamesTheAwesomeDude** [comment]
> As we have only adapted the upstream source code, this is a question for the Classic McEliece submission team.

This definitely canonically closes this ticket, anyway. I checked out the upstream source and it's only slightly less scrutable than this codebase.
